### ISSUE TITLE
Cypress 4.0.0 broke the matching of non-standard spaces

### ISSUE THREAD

**Issue by codeheroics** [cypress-io/cypress#6959]
### Current behavior:

Cypress 4.0.0 introduced a breaking change intended to better match browsers behaviour and [ignore invisible whitespaces](https://docs.cypress.io/guides/references/migration-guide.html#cy-contains-ignores-invisible-whitespaces).

This has, however, broken the ability to match non standard spaces, such as non-breaking spaces (and I guess other types of states).

To further explain my use-case : I'm loading the same data for wording in my tests and my production code, which allows me to ensure using for example `cy.contains(wordingForEmailWarning)` that those strings are correctly displayed in the page. The same `wordingForEmailWarning` variable, containing the exact same string, is then used in my application code.

Starting with Cypress 4.0.0 however, any string given to `cy.contain()` containing a non-breaking space fails to match the displayed string, because of the new white-space matching rules. This is particularly inconvenient because to work, Cypress would require me to rewrite the strings I'm trying to match using only simple white-spaces`, which would be wrong (non-breaking white spaces are used in a lot of languages, before punctuation for example).

### Desired behavior:

Allow matching the strings such as the one given in the test repo below. The matching of Cypress 3.x allowed this.

### Test code to reproduce
[Test repository](http://github.com/codeheroics/cypress-test-tiny), which points to [this page](https://codeheroics.github.io/nbsp/), which contains a sentence with words only separated with non breaking spaces.

This issue is also related to https://github.com/cypress-io/cypress/issues/6405 : The changes to the space handling of Cypress 4.0.0 are what introduce this error, and cause other problems for the use-case previously explained.

**Comment by jennifer-shehane** [comment]
Running the test code provided passes on both tests. Can you provide a failing test case so that I can see the problem exactly that you're explaining?

```js
describe("page", () => {
  it("does not work with the original string", () => {
    cy.visit("https://codeheroics.github.io/nbsp/");
    cy.contains("I am text containing nonbreaking spaces which are important in some languages for punctuation");
  });
  it("works with all spaces replaced with normal spaces inside the test", () => {
    cy.visit("https://codeheroics.github.io/nbsp/");
    cy.contains("I am text containing nonbreaking spaces which are important in some languages for punctuation");
  });
});
```

<img width="921" alt="Screen Shot 2020-04-08 at 11 18 52 AM" src="https://user-images.githubusercontent.com/1271364/78745622-beb1ea00-798a-11ea-9d9b-c7f527681095.png">

Also I'm not sure how this is not a duplicate of #6405 entirely.

**Comment by codeheroics** [comment]
![Screenshot from 2020-04-08 10-43-13](https://user-images.githubusercontent.com/1598317/78764054-4c3f0b00-7986-11ea-9cb4-50368701db7d.png)

Sorry, but this is my failing test case, I can't really provide you anything better :cold_sweat: the unmodified repository crashes on my machine on Chrome, Firefox and Electron, both when using `cypress open` and `cypress run`.

I guess there must be something special with my configuration, perhaps the OS since I'm using Linux? I don't have access to a Mac computer to check how it works

For the differences with #6405, (assuming the issue can actually be reproduced), I'd argue that the string `Questions ?`, with the word and the question mark separated by a non-breaking space (not the case in this comment), if provided exactly this way in both `.contains` and HTML code, should match without issue, unrelated to the treatment of normal spaces. On my computer, I'm unable to match these characters *at all*.

**Comment by jennifer-shehane** [comment]
The PR for that issue https://github.com/cypress-io/cypress/pull/6411 will allow an option to `matchWhitespace` when provided. Wouldn't that meet what you are looking for?

```js
cy.get('#btn').contains('  name    age   ', { matchWhitespaces: true })
```

**Comment by codeheroics** [comment]
It may fix it as a side-effect, but it would be just that, a side-effect. In this particular case, I don't want to opt out of space collapsing, I want this non-breaking space to be matched correctly. Since it happens even with a single space-like character, I don't see why I'd need to tell Cypress to use special rules to match it

I'll represent for this example the nonbreaking white-space character with `" "`. Assuming the HTML contains a ```Any   questions" "?``` string, with 3 normal spaces between "Any" and "questions", which when displayed are collapsed into one:
```js
// should pass
cy.get('#btn').contains('Any questions" "?')
 // should not pass, but would pass with `matchWhiteSpaces`
cy.get('#btn').contains('Any    questions" "?')
```


Also, related to my [previous comment](https://github.com/cypress-io/cypress/issues/6959#issuecomment-610841302), I've had a friend using MacOS run the tests for me, and they came back failing as well, I don't get why they passed when you tried them :slightly_frowning_face:

**Comment by egrubbs** [comment]
I have also noticed that the Cypress Command Log makes it impossible to tell that element contains 
`"\xa0"` (i.e. &nbsp;) instead of `" "`. 
The error message will basically say that `" " should equal " "` which is incredibly confusing.
```
cy.get('div#foo').invoke('text').should('equal', ' ');
```
If I try to log the variables myself, I get the same problem.  If I copy and paste the text out of the command log and run `escape()` on it, I see that `"\a0"` has been replaced with `" "`.
```
cy.get('div#foo').invoke('text').then((text) => {
  cy.log(`${JSON.stringify(text)} should equal " ": ${text === " "}`);
});
```
I end up having to use ```cy.log(`${escape(text)}`)``` just so I can tell that the text inside the element is `"\xa0"` instead of a blank space.
